src/AstromFit.cc

```diff
@@ -947,295 +947,298 @@ static void write_vect_in_fits(const Eigen::VectorXd &V, const string &FitsName)
-  // 1 : still some ouliers but chi2 increases
-  // 2 : factorization failed
-  unsigned returnCode = 0;
-
-  // TODO : write a guesser for the number of triplets
-  unsigned nTrip = (_LastNTrip) ? _LastNTrip: 1e6;
-  TripletList tList(nTrip);
-  Eigen::VectorXd grad(_nParTot);  grad.setZero();
-
-  //Fill the triplets
-  clock_t tstart = clock();
-  LSDerivatives(tList, grad);
-  clock_t tend = clock();
-  _LastNTrip = tList.size();
-
-  cout << " INFO: End of triplet filling, ntrip = " << tList.size()
-       << " CPU = " << float(tend-tstart)/float(CLOCKS_PER_SEC)
-       << endl;
-
-  SpMat hessian;
-  {
-#if (TRIPLET_INTERNAL_COORD == COL)
-    SpMat jacobian(_nParTot,tList.NextFreeIndex());
-    jacobian.setFromTriplets(tList.begin(), tList.end());
-    // release memory shrink_to_fit is C++11
-    tList.clear(); //tList.shrink_to_fit();
+    assignIndices(whatToFit);
+
+    // return code can take 3 values :
+    // 0 : fit has converged - no more outliers
+    // 1 : still some ouliers but chi2 increases
+    // 2 : factorization failed
+    unsigned returnCode = 0;
+
+    // TODO : write a guesser for the number of triplets
+    unsigned nTrip = (_LastNTrip) ? _LastNTrip : 1e6;
+    TripletList tList(nTrip);
+    Eigen::VectorXd grad(_nParTot);  grad.setZero();
+
+    //Fill the triplets
     clock_t tstart = clock();
-    hessian = jacobian*jacobian.transpose();
+    LSDerivatives(tList, grad);
     clock_t tend = clock();
-    std::cout << "INFO: CPU for J*Jt "
-	      << float(tend-tstart)/float(CLOCKS_PER_SEC) << std::endl;
+    _LastNTrip = tList.size();
+
+    cout << " INFO: End of triplet filling, ntrip = " << tList.size()
+         << " CPU = " << float(tend - tstart) / float(CLOCKS_PER_SEC)
+         << endl;
+
+    SpMat hessian;
+    {
+#if (TRIPLET_INTERNAL_COORD == COL)
+        SpMat jacobian(_nParTot, tList.NextFreeIndex());
+        jacobian.setFromTriplets(tList.begin(), tList.end());
+        // release memory shrink_to_fit is C++11
+        tList.clear(); //tList.shrink_to_fit();
+        clock_t tstart = clock();
+        hessian = jacobian * jacobian.transpose();
+        clock_t tend = clock();
+        std::cout << "INFO: CPU for J*Jt "
+                  << float(tend - tstart) / float(CLOCKS_PER_SEC) << std::endl;
 
 #else
-    SpMat jacobian(tList.NextRank(), _nParTot);
-    jacobian.setFromTriplets(tList.begin(), tList.end());
-    // release memory shrink_to_fit is C++11
-    tList.clear(); //tList.shrink_to_fit();
-    cout << " starting H=JtJ " << endl;
-    hessian = jacobian.transpose()*jacobian;
+        SpMat jacobian(tList.NextRank(), _nParTot);
+        jacobian.setFromTriplets(tList.begin(), tList.end());
+        // release memory shrink_to_fit is C++11
+        tList.clear(); //tList.shrink_to_fit();
+        cout << " starting H=JtJ " << endl;
+        hessian = jacobian.transpose() * jacobian;
 #endif
-  }// release the Jacobian
+    }// release the Jacobian
```

Comment:
I interpret this comment to mean that you free the memory from 'jacobian', but I don't see that happening.